2011-04-17  Ulrich Drepper  <drepper@gmail.com>

	[BZ #12420]
	* sysdeps/unix/sysv/linux/x86_64/getcontext.S: Reload context after
	storing it.

---
 sysdeps/unix/sysv/linux/x86_64/getcontext.S |    1 +
 1 file changed, 1 insertion(+)

--- a/sysdeps/unix/sysv/linux/x86_64/getcontext.S
+++ b/sysdeps/unix/sysv/linux/x86_64/getcontext.S
@@ -62,6 +62,7 @@
 	movq	%rcx, oFPREGS(%rdi)
 	/* Save the floating-point environment.  */
 	fnstenv	(%rcx)
+	fldenv	(%rcx)
 	stmxcsr oMXCSR(%rdi)
 
 	/* Save the current signal mask with
